File: clear.py
#!/usr/bin/env python

import rospy
from std_srvs.srv import Empty

def clear_costmaps():
    rospy.wait_for_service('/move_base/clear_costmaps')
    try:
        clear_costmaps_service = rospy.ServiceProxy('/move_base/clear_costmaps', Empty)
        clear_costmaps_service()
        rospy.loginfo("Costmaps cleared successfully.")
    except rospy.ServiceException as e:
        rospy.logerr("Service call failed: %s" % e)

def main():
    rospy.init_node('navigation_with_clear_costmaps')
    
    # Example navigation task
    while not rospy.is_shutdown():
        # Perform navigation tasks here
        # ...
        
        # Clear costmaps periodically
        clear_costmaps()
        rospy.sleep(2)  # Adjust the sleep time as needed

if __name__ == "__main__":
    main()

File: Jetsonnano/catkin_ws/src/clear/src/clear.cpp
#include <ros/ros.h>
#include <std_srvs/Empty.h>

void clearCostmaps()
{
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<std_srvs::Empty>("/move_base/clear_costmaps");
    std_srvs::Empty srv;

    if (client.call(srv))
    {
        ROS_INFO("Costmaps cleared successfully.");
    }
    else
    {
        ROS_ERROR("Failed to call service clear_costmaps");
    }
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "navigation_with_clear_costmaps_cpp");
    ros::NodeHandle nh;

    // Example navigation task
    while (ros::ok())
    {
        // Perform navigation tasks here
        // ...

        // Clear costmaps periodically
        clearCostmaps();
        ros::Duration(2).sleep(); // Adjust the sleep time as needed
    }

    return 0;
}

